Realm/Realm/Sync/MongoClient.cs

```diff
@@ -162,10 +170,17 @@ public class Collection<TDocument>
 
             internal Collection(Database database, string name)
             {
+                _client = database.Client;
                 Database = database;
                 Name = name;
             }
 
+            internal Collection(MongoClient client, string name)
```

Comment:
I kept the same collection type for both `client.GetCollection` and `client.Database.GetCollection`, as the property names (and descriptions) seems to be compatible enough.